clippy_lints/src/doc/suspicious_doc_comments.rs

```diff
@@ -32,12 +33,14 @@ pub fn check(cx: &LateContext<'_>, attrs: &[Attribute]) -> bool {
     }
 }
 
-fn collect_doc_replacements(attrs: &[Attribute]) -> Vec<(Span, String)> {
+fn collect_doc_replacements(cx: &LateContext<'_>, attrs: &[Attribute]) -> Vec<(Span, String)> {
     attrs
         .iter()
         .filter_map(|attr| {
             if let Some((sym, com_kind)) = attr.doc_str_and_comment_kind()
```

Comment:
So it seems like HIR attributes still differentiate between `///` (`AttrKind::DocComment`) and `#[doc]` (`AttrKind::Normal`), and the lint did actually handle this correctly until this was recently changed to the more general `.doc_str_and_comment_kind()` here.

@jdonszelmann do you intend to get rid of `AttrKind::DocComment` in your HIR attribute refactor or can we just change this line back to `if let AttrKind::DocComment(sym, com_kind) = attr.kind` like before?

Reply:
So here, .doc_comment_kind() still works but mostly to make it so less code needs to change. But in the future we should transition to matching on rustc_attr_data_structures::AttributeKind.

Reply:
Though, and I'm sorry if this is true, specifically for doc comments there may need to be a little more logic in the compiler so you could choose to keep using this method and I'll replace it some day.

Reply:
I'll adjust the PR after the rustup to only keep the test.